src/Files.App/Utils/Git/GitHelpers.cs

```diff
@@ -315,14 +366,25 @@ public static async Task PullOriginAsync(string? repositoryPath)
 					};
 			}
 
-			IsExecutingGitAction = true;
+			MainWindow.Instance.DispatcherQueue.TryEnqueue(() =>
+			{
+				IsExecutingGitAction = true;
+			});
 
 			try
 			{
-				LibGit2Sharp.Commands.Pull(
-					repository,
-					signature,
-					_pullOptions);
+				Interlocked.Increment(ref _activeOperationsCount);
+				_owningThread ??= new ThreadWithMessageQueue();
+
+				await _owningThread.PostMethod(() =>
+				{
+					LibGit2Sharp.Commands.Pull(
+						repository,
+						signature,
+						_pullOptions);
+				});
+
+				DisposeIfFinished();
```

Comment:
Sorry to keep pointing out.
These processes can be combined in one place by creating a method that takes an Action as an argument like following.
```csharp
private static async Task PostMethodToThreadWithMessageQueue(Action payload)
{
	Interlocked.Increment(ref _activeOperationsCount);
	_owningThread ??= new ThreadWithMessageQueue();

	await _owningThread.PostMethod(payload);

	DisposeIfFinished();
}
```